SofaKernel/modules/SofaCore/src/sofa/core/topology/TopologyHandler.cpp

```diff
@@ -50,6 +55,71 @@ void TopologyHandler::ApplyTopologyChanges(const std::list<const core::topology:
         std::string topoChangeType = "DefaultTopologyHandler: " + parseTopologyChangeTypeToString(changeType);
         sofa::helper::AdvancedTimer::stepBegin(topoChangeType);
 
+
+        // New version using map of callback
+        std::map < core::topology::TopologyChangeType, TopoChange_callback>::iterator itM;
+        itM = m_callbackMap.find(changeType);
+
+        if (itM != m_callbackMap.end())
+        {
+            std::cout << "m_callbackMap found" << std::endl;
+            (*itM).second(*changeIt);
```

Comment:
I prefer this approach than the switch/case. I suggest to remove the commented code and remove the macro defined earlier.